python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -284,6 +315,99 @@ def _process_adapter_response_metadata(
 
         return processed_adapter_response
 
+    def _process_lineage_metadata(
+        self,
+        manifest: Mapping[str, Any],
+        dagster_dbt_translator: DagsterDbtTranslator,
+        target_path: Path,
+    ) -> Dict[str, Any]:
+        """Process the lineage metadata for a dbt CLI event.
+
+        Args:
+            manifest (Mapping[str, Any]): The dbt manifest blob.
+            dagster_dbt_translator (DagsterDbtTranslator): The translator for dbt nodes to Dagster assets.
+            target_path (Path): The path to the dbt target folder.
+
+        Returns:
+            Dict[str, Any]: The lineage metadata.
+        """
+        # TODO: this should be refactored as a Python object that renders the JSON metadata.
+        lineage_metadata: Dict[str, List[Tuple[AssetKey, str]]] = {}
+
+        event_node_info: Dict[str, Any] = self.raw_event["data"].get("node_info")
+        unique_id: str = event_node_info["unique_id"]
+        dbt_resource_props: Dict[str, Any] = manifest["nodes"][unique_id]
+
+        # If the unique_id is a seed or source, then we don't need to process lineage.
+        if unique_id.startswith("seed") or unique_id.startswith("source"):
+            return {}
+
+        # 1. Retrieve all of the node ancestors and their SQL files (if they exist)
```

Comment:
I'm having trouble squaring this comment with what I'm seeing in the code below. Where do we retrieve the node ancestor SQL files?

Reply:
Oh is the idea that it will already have been placed into `sqlglot_sources` during a previous invocation of this function? Confusing to follow, but maybe will be solved as a byproduct when you find a way to make that not global?

Reply:
One thing to think about here is that, if they're only executing a subset of models, that might not be populated?

Reply:
See https://github.com/dagster-io/dagster/pull/20407/files#r1525589805